### DISCUSSION TITLE
Is there a good way to locate the core dump caused by JS problems?

### DISCUSSION THREAD

**Discussion by yudaosi** [quickjs-ng/quickjs#772] [Q&A]
The quickjs is used to run the JS service system code. There is a possibility that coredump occurs when the JS code is running. No exception caused by the service c code is found in the coredump file. All exceptions are found in the libc.so file. No error is found in ref_count and no obvious JS code execution error is found. 
We found if JS code has some error, After the quickjs engine is executed for a long period of time, coredumps may occur on the quickjs engine. 
Please explain the cause of the coredump caused by the JS code.
How to locate the problem and can this be avoided?
Thank you!

**Comment by bnoordhuis** [comment]
Converted to a discussion because your bug report is too broad and vague to be actionable, it's basically "it doesn't work, what should I do to fix it?"

Try the regular things: look at the stack traces in the core dumps, turn on sanitizers (ASan, UBSan, MSan) and see what they report, etc.

**Reply by yudaosi** [reply]
thinks reply,
Does the Quickjs engine have a debug mode to generate a report when a problem occurs?

**Reply by saghul** [reply]
Compiling in debug mode does not generate a report. The generated core dumps will have symbols though.
